be/src/vec/data_types/serde/data_type_datetimev2_serde.cpp

```diff
@@ -115,10 +117,24 @@ void DataTypeDateTimeV2SerDe::write_column_to_arrow(const IColumn& column, const
             if (scale > 3) {
                 uint32_t microsecond = datetime_val.microsecond();
                 timestamp = (timestamp * 1000000) + microsecond;
+                timestamp -= std::chrono::duration_cast<std::chrono::microseconds>(
+                                     cctz::convert(cctz::civil_second(1970, 1, 1, 0, 0, 0), ctz)
+                                             .time_since_epoch())
+                                     .count();
             } else if (scale > 0) {
                 uint32_t millisecond = datetime_val.microsecond() / 1000;
                 timestamp = (timestamp * 1000) + millisecond;
+                timestamp -= std::chrono::duration_cast<std::chrono::milliseconds>(
+                                     cctz::convert(cctz::civil_second(1970, 1, 1, 0, 0, 0), ctz)
+                                             .time_since_epoch())
+                                     .count();
+            } else if (scale == 0) {
+                timestamp -= std::chrono::duration_cast<std::chrono::seconds>(
```

Comment:
Should the logic here be removed? It should only require modifying the scale of the TypeDescriptor.

Reply:
> Should the logic here be removed? It should only require modifying the scale of the TypeDescriptor.

Already Reverted